src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -287,17 +288,26 @@ void mp::LibVirtVirtualMachine::stop()
 
 void mp::LibVirtVirtualMachine::shutdown()
 {
+    std::unique_lock<decltype(state_mutex)> lock{state_mutex};
     if (state == State::running || state == State::delayed_shutdown)
     {
         virDomainShutdown(domain.get());
         state = State::off;
         update_state();
     }
+    else if (state == State::starting)
+    {
+        virDomainDestroy(domain.get());
```

Comment:
Because we can assume if a `stop` is issued while an instance is starting, the we can just "pull the plug" on the VM and not worry about a graceful shutdown, which is what `virDomainShutdown()` does- a graceful shutdown.  In fact, if `virDomainShutdown()` is issued while the instance is booting, it is ignored.

Reply:
It's akin to pulling the plug on a physical machine during boot.  You can then push the power button again and it will boot.

Reply:
"Pull plug" made me worry, but libvirt does try to get the VM to shutdown properly:
https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainDestroy
So this is good enough for me.